services/blockchain-connector/shared/sdk/genesisBlock.js

```diff
@@ -83,9 +90,81 @@ const getGenesisConfig = async () => {
 	}
 };
 
+const getGenesisAssets = async (params = {}) => {
+	const genesisBlock = await getGenesisBlockFormatted(true);
+
+	// Return all genesis block assets if no module / subStore key present in params
+	if (!params.module && !params.subStore) return genesisBlock.assets;
+
+	const assetByModule = genesisBlock.assets.find(asset => asset.module === params.module);
+
+	// Filter data by subStore if passed in input
+	if (params.subStore) {
+		let moduleData = assetByModule.data[params.subStore];
+
+		// Return empty array otherwise if passed subStore is not present
+		if (!moduleData) return [];
+
+		// Filter module data based on limit and offset
+		if (typeof params.offset !== 'undefined' && params.limit) {
+			moduleData = moduleData.slice(params.offset, params.offset + params.limit);
+		}
+
+		return [{
+			...assetByModule,
+			data: {
+				[params.subStore]: moduleData,
+			},
+		}];
+	}
```

Comment:
May be we can update this and avoid using `let` if possible. Also, `limit` and `offset` will be always available as we are setting default values in controller
```suggestion
		if (params.subStore) {
		// Return empty array otherwise if passed subStore is not present
		const moduleData = assetByModule.data[params.subStore]
			? assetByModule.data[params.subStore]
			: [];

		return [{
			...assetByModule,
			data: {
				[params.subStore]: moduleData.slice(params.offset, params.offset + params.limit),
			},
		}];
	}
```

Reply:
limit and offset will not be present always. ex: https://github.com/LiskHQ/lisk-service/blob/1546-incorrect-total-locked-and-total-self-staked-values/services/blockchain-connector/shared/sdk/genesisBlock.js#L148